libsolidity/analysis/TypeChecker.cpp

```diff
@@ -113,18 +113,24 @@ void TypeChecker::checkDoubleStorageAssignment(Assignment const& _assignment)
 	size_t storageByteArrayPushes = 0;
 	size_t storageByteAccesses = 0;
 	auto count = [&](TupleExpression const& _lhs, TupleType const& _rhs, auto _recurse) -> void {
-		TupleType const& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
-		TupleExpression const* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
+		auto& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
+		auto* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
+
+		if (!lhsResolved) {
+			m_errorReporter.reportErrors("lhsResolved is null");
+			return;
+		}
```

Comment:
If this change is correct, we should see that in tests.

Please add the test case from the issue so that we can see it works now.

I'd also recommend adding some variations of the case to make sure this syntax is well covered.

Reply:
Thank you for the review. Which library I need using for writing tests in C++: gtest or another? Alternatively, should I write tests on Solidity

Reply:
We have a custom tool for testing syntax/semantics via small code snippets. Take a look at the [`syntaxTests`](https://github.com/ethereum/solidity/tree/develop/test/libsolidity/syntaxTests) dir. It should be pretty self-explanatory. See [Running the Compiler Tests](https://docs.soliditylang.org/en/latest/contributing.html#running-the-compiler-tests) in the docs on how to run these tests locally.